workspace/files/tests/test_webdav_mount.py

```
@@ -0,0 +1,580 @@
+        )
+
+    def test_cp_duplicates_file(self):
+        """``cp a b`` → COPY → both rows live, content matches."""
+        src = self.mp / "orig.txt"
+        src.write_bytes(b"copy me")
+        self._wait_for_file(owner=self.user, name="orig.txt")
+
+        # cp shells out by default; subprocess keeps cp's exact behavior
+        # (some clients implement copy as GET + PUT, others as COPY —
+        # GNU cp on a davfs2 mount goes through the COPY method).
+        r = subprocess.run(
```

Ruff (0.15.11)

[error] `subprocess` call: check for execution of untrusted input

(S603)